src/enclave/enclave_init.c

```diff
@@ -89,7 +89,29 @@ static void init_wireguard()
         wgu_list_devices();
 }
 
-static int startmain(void* args)
+static int app_main_thread(void* args)
+{
+    SGXLKL_VERBOSE("enter\n");
+
+    lthread_set_funcname(lthread_self(), "app-main");
+    lthread_set_app_main();
```

Comment:
Can this be done in the caller?  We're mixing lthread and pthread calls here, which is going to make finishing the layering cleanup harder.  All of this function should be considered part of userspace.  It makes me slightly unhappy that this thread is created with `lthread_create` and not a clone `lkl_syscall`, but mixing lthread and pthread calls within the same function is definitely not something we should be doing.